src/Native/System.Native/pal_networking.cpp

```diff
@@ -2327,32 +2332,37 @@ static Error TryChangeSocketEventRegistrationInner(
     bool writeChanged = (changes & PAL_SA_WRITE) != 0;
 
     struct kevent events[2];
+    auto event = &events[0];
+    int i = 1;
+
+    event->ident = static_cast<uint64_t>(socket);
+    event->fflags = 0;
+    event->data = 0;
+#if KEVENT_HAS_VOID_UDATA
+    event->udata = reinterpret_cast<void*>(data);
+#else
+    event->udata = static_cast<intptr_t>(data);
+#endif
```

Comment:
umm, didn't you ask me to expand the macro: https://github.com/dotnet/corefx/pull/6057#discussion_r52771130 ?